### PR TITLE
mocap_pose plugin changed to work with mocap4ros2

### PR DESCRIPTION
**Pull request by pariaspe**
## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Issue(s) this addresses   | #270 |
| ROS2 version tested on | Humble |
| Aerial platform tested on | Crazyflie |

---

## Description of contribution in a few bullet points

* mocap_pose plugin changed to work with [mocap4ros2](https://github.com/MOCAP4ROS2-Project)
* Added orientation smoother, needed to stable flights with Vicon
- **EDIT**: Added parameters for smooth filter constants (speed and orientation). Default values in launcher are `speed_alpha=0.1` and `orientation_alpha=1.0` (filter disabled). If launcher is not used and node is executed standalone both filters will be disabled by default (`aplha=1.0`).

## Description of documentation updates required from your changes

* Since [aerostack2/mocap_optitrack](https://github.com/aerostack2/mocap_optitrack) will be replaced with [mocap4ros2](https://github.com/MOCAP4ROS2-Project), documentation should link how to launch a mocap4ros2 node

---

## Future work that may be required in bullet points

* Tested on a Vicon system only. **Testing this with an Optitrack system before merging might be required**
* ~~New parameter for orientation smoother -> `alpha`. If `alpha=1.0`, smoother will be disable.~~

### CODE REVIEW

**Comment by pariaspe** [comment]
PR frozen, waiting for https://github.com/MOCAP4ROS2-Project/mocap_msgs/issues/11 and https://github.com/MOCAP4ROS2-Project/mocap4ros2_vicon/pull/17

File: as2_state_estimator/plugins/mocap_pose/include/mocap_pose.hpp [hunk 1]
```diff
@@ -190,12 +191,24 @@ class Plugin : public as2_state_estimator_plugin_base::StateEstimatorBase {
     odom_to_base_msg.child_frame_id  = get_base_frame();
     publish_transform(odom_to_base_msg);
 
+    // Orientation smoother
+    const double alpha = 0.1;
```
Comment (miferco97) [inline as2_state_estimator/plugins/mocap_pose/include/mocap_pose.hpp]: can this be a plugin param ?

Comment (miferco97) [reply as2_state_estimator/plugins/mocap_pose/include/mocap_pose.hpp]: Oh, Sorry :S

**Comment by miferco97** [comment]
I think that we should wait for [#MOCAP4ROS2/mocap_msgs/Issue11](https://github.com/MOCAP4ROS2-Project/mocap_msgs/issues/11#issue-1972685265) to answer us before merging.

**Comment by pariaspe** [comment]
> I think that we should wait for [#MOCAP4ROS2/mocap_msgs/Issue11](https://github.com/MOCAP4ROS2-Project/mocap_msgs/issues/11#issue-1972685265) to answer us before merging.

Yes, I completely agree. CI is not working due to missing mocap_msgs dependency

**Comment by miferco97** [comment]
@fmrico @jmguerreroh could it be possible to release mocap_msgs package into rosdistro? If the project is not maintained anymore please let us know in order to search for alternative solutions. Thanks in advance.

**Comment by jmguerreroh** [comment]
> @fmrico @jmguerreroh could it be possible to release mocap_msgs package into rosdistro? If the project is not maintained anymore please let us know in order to search for alternative solutions. Thanks in advance.

Hi @miferco97 @pariaspe, 

We're sorry for the delay and assure you that mocap4ros2's inclusion in rosdistro is actively being worked on. We will let you know of any updates. 

Thank you for your patience and support :)

**Comment by miferco97** [comment]
@pariaspe @javilinos @RPS98 I think we should merge this and update our projects accordingly and test it. What do you think?

**Comment by pariaspe** [comment]
> @pariaspe @javilinos @RPS98 I think we should merge this and update our projects accordingly and test it. What do you think?

Sure! Javi and I already did some updates in project_crazyflie and gates. I'll open a PR later on those repos

**Comment by RPS98** [comment]
> @pariaspe @javilinos @RPS98 I think we should merge this and update our projects accordingly and test it. What do you think?

Sure! I think we should merge the PRs for aerostack2, the projects mentioned in the documentation, and the documentation itself synchronously.

On the other hand, is it worth maintaining support for the old one? We could use templates to make it easier to change the message type.

**Review by javilinos — COMMENTED** [review]
Good to go

**Comment by RPS98** [comment]
Hi @javilinos,

Why are you merging this branch if the project's and documentation PRs aren't open?

**Comment by javilinos** [comment]
> Hi @javilinos,
> 
> Why are you merging this branch if the project's and documentation PRs aren't open?

Hi @RPS98 why is this branch not in draft if it cannot be merged?

**Comment by javilinos** [comment]
> > Hi @javilinos,
> > Why are you merging this branch if the project's and documentation PRs aren't open?
> 
> Hi @RPS98 why is this branch not in draft if it cannot be merged?
@RPS98 
I should have read deeper, I'm sorry, we have two public CF projects, we will make PRs asap :)

(for next time if we want to discuss on a branch that is not ready to be merged we should draft it)

**Pull request merged**